packages/react-native-renderer/src/ReactFabric.js

```diff
@@ -63,9 +63,11 @@ function findNodeHandle(componentOrHandle: any): ?number {
   if (hostInstance == null) {
     return hostInstance;
   }
-  if (hostInstance.canonical) {
+  // TODO: the code is right but the types here are wrong.
+  // https://github.com/facebook/react/pull/12863
+  if ((hostInstance: any).canonical) {
```

Comment:
I'll be working on a followup to fix this. The types are already wrong due to reconciler `findHostInstance` typing.